modules/reitit-openapi/src/reitit/openapi.cljc

```diff
@@ -0,0 +1,118 @@
+  | :description   | optional long description of an endpoint. Supports http://spec.commonmark.org/
+
+  Also the coercion keys contribute to openapi spec:
+
+  | key           | description |
+  | --------------|-------------|
+  | :parameters   | optional input parameters for a route, in a format defined by the coercion
+  | :responses    | optional descriptions of responses, in a format defined by coercion
+
+  Use `:request` parameter coercion (instead of `:body`) to unlock per-content-type coercions.
+
+  Example:
+
+      [\"/api\"
+       {:openapi {:id :my-api}
+        :middleware [reitit.openapi/openapi-feature]}
+
+       [\"/openapi.json\"
+        {:get {:no-doc true
+               :openapi {:info {:title \"my-api\"}}
+               :handler reitit.openapi/openapi-spec-handler}}]
+
+       [\"/plus\"
+        {:get {:openapi {:tags \"math\"}
+               :summary \"adds numbers together\"
+               :description \"takes `x` and `y` query-params and adds them together\"
+               :parameters {:query {:x int?, :y int?}}
+               :responses {200 {:body {:total pos-int?}}}
+               :handler (fn [{:keys [parameters]}]
+                          {:status 200
+                           :body (+ (-> parameters :query :x)
+                                    (-> parameters :query :y)})}}]]"
+  {:name ::openapi
+   :spec ::spec})
+
+(defn- openapi-path [path opts]
+  (-> path (trie/normalize opts) (str/replace #"\{\*" "{")))
+
+(defn create-openapi-handler
+  "Stability: alpha
+
+  Create a ring handler to emit openapi spec. Collects all routes from router which have
+  an intersecting `[:openapi :id]` and which are not marked with `:no-doc` route data."
+  []
+  (fn create-openapi
+    ([{::r/keys [router match] :keys [request-method]}]
+     (let [{:keys [id] :or {id ::default} :as openapi} (-> match :result request-method :data :openapi)
+           ids (trie/into-set id)
+           strip-top-level-keys #(dissoc % :id :info :host :basePath :definitions :securityDefinitions)
+           strip-endpoint-keys #(dissoc % :id :parameters :responses :summary :description)
+           openapi (->> (strip-endpoint-keys openapi)
+                        (merge {:openapi "3.1.0"
+                                :x-id ids}))
+           accept-route (fn [route]
+                          (-> route second :openapi :id (or ::default) (trie/into-set) (set/intersection ids) seq))
+           ;base-openapi-spec {:responses ^:displace {:default {:description ""}}}
+           transform-endpoint (fn [[method {{:keys [coercion no-doc openapi] :as data} :data
+                                            middleware :middleware
+                                            interceptors :interceptors}]]
+                                (if (and data (not no-doc))
+                                  [method
+                                   (meta-merge
+                                    #_base-openapi-spec
```

Comment:
why is this commented out?

Reply:
no idea, left over from the previous PR #563. deleted.